perf/perf_demangle.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2025 IBM
+# Author: Tejas Manhas <Tejas.Manhas1@ibm.com>
+
+import os
+import re
+import shutil
+from avocado import Test
+from avocado.utils import distro, process, genio, cpu
+from avocado.utils.software_manager.manager import SoftwareManager
+from avocado.utils.ssh import Session
+
+
+class demangle(Test):
+    """
+    This is a test class for demangle feature in perf record report for C++ workloads.
+    """
+
+    def setUp(self):
+        '''
+        Install the basic packages to support perf
+        '''
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        if 'ppc64' not in detected_distro.arch:
+            self.cancel('This test is not supported on %s architecture'
+                        % detected_distro.arch)
+        if 'PowerNV' in genio.read_file('/proc/cpuinfo'):
+            self.cancel('This test is only supported on LPAR')
+        process.run("dmesg -C")
+        deps=["cmake"]
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+    def run_cmd(self, cmd):
+        """
+        run command on SUT as root
+        """
+        result = process.run(cmd, shell=True, ignore_status=True)
+        output = result.stdout_text + result.stderr_text
+        return output
+
+    def install_google_benchmark(self):
+        self.run_cmd("git clone https://github.com/google/benchmark.git")
+        os.chdir("benchmark")
+        self.run_cmd("git clone https://github.com/google/googletest.git")
```

Comment:
utils command exist to clone the repo.. check for fetch_url

Reply:
you can also use git.get_repo